docs/source/design/clis.rst

```diff
@@ -22,14 +22,67 @@ The client code is located in ``flytekit/clients`` and there are two.
+
+It is considered fast registration because when a script is executed using ``pyflyte run``, the script is bundled up and uploaded to FlyteAdmin. When the task is executed in the backend, this zipped file is extracted and used.
+
+``pyflyte run`` produces a **portable** package (a .tgz file) of Flyte entities, which helps introspect them at a later time too. You can register this package with multiple Flyte backends. You can save this package, use it for audit, register with a different admin, and so on.
+
+Usage
+=====
+
+.. prompt:: bash $
+
+   pyflyte --config ~/.flyte/dev-uniondemo.yaml register --image ghcr.io/flyteorg/flytecookbook:core-latest --image trainer=ghcr.io/flyteorg/flytecookbook:core-latest --image predictor=ghcr.io/flyteorg/flytecookbook:core-latest --raw-data-prefix s3://development-service-flyte/reltsts flyte_basics
+
+.. note ::
+
+   Neither ``pyflyte register`` nor ``pyflyte run`` commands work on Python namespace packages since both the tools traverse the filesystem to find the first folder that doesn't have an __init__.py file, which is interpreted as the root of the project. Both the commands use this root as the basis to name the Flyte entities.
+
+In a broad way, ``pyflyte register`` is equivalent to ``pyflyte run`` minus launching workflows, with the exception that ``pyflyte run`` can only register a single workflow, whereas ``pyflyte register`` can register all workflows in a repository.
+
+Why should you use ``pyflyte register``?
+========================================
+
+The ``pyflyte register`` command bridges the gap between ``pyflyte package`` + ``flytectl register``and ``pyflyte run`` commands. It offers the functionality of the ``pyflyte package`` (with smarter naming semantics and combines the network call into one step), by only registering all the workflows in the repository.
+
+.. note ::
+
+   You can’t use ``pyflyte register`` if you are unaware of the run-time options yet(IAM role, service account, and so on).
+
+Should I use ``pyflyte run`` or ``pyflyte package`` + ``flytectl register``?
+============================================================================
+
+Both the commands have their own place in a production Flyte setting.
+
+``pyflyte run`` is convenient when you are starting out with Flyte, testing small scripts, or iterating over local scripts.
 
-This CLI is what is responsible for the serialization (compilation) step in the registration flow. It will parse through user code, looking for tasks, workflows and launch plans, and compile them down to Protobuf files.
+``pyflyte package`` + ``flytectl register`` is useful for single admin use-cases where you want to zip your files (that constitute all your tasks and workflows) into a single portable file and pass it on to another developer/location to register it or run something in CI.
```

Comment:
Not sure what is meant by single admin use-case here.  that comment is for pyflyte register, not package.  making a portable package implies multiple admins.